stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -33,18 +46,276 @@ public void disable(JsonRpcPeer peer, JSONObject params) {
   }
 
   @ChromeDevtoolsMethod
-  public JsonRpcResult getSupportedCSSProperties(JsonRpcPeer peer, JSONObject params) {
-    GetSupportedCSSPropertiesResponse response = new GetSupportedCSSPropertiesResponse();
-    response.cssProperties = Collections.emptyList();
-    return response;
+  public JsonRpcResult getComputedStyleForNode(JsonRpcPeer peer, JSONObject params) {
+    final GetComputedStyleForNodeRequest request = mObjectMapper.convertValue(
+        params,
+        GetComputedStyleForNodeRequest.class);
+
+    final GetComputedStyleForNodeResult result = new GetComputedStyleForNodeResult();
+    result.computedStyle = new ArrayList<>();
+
+    mDocument.postAndWait(new Runnable() {
+      @Override
+      public void run() {
+        mDocument.getElementStyle(
+            mDocument.getElementForNodeId(request.nodeId),
+            new StyleAccumulator() {
+              @Override
+              public void store(String name, String value, boolean isDefault) {
+                if (!isDefault) {
+                  CSSComputedStyleProperty property = new CSSComputedStyleProperty();
+                  property.name = name;
+                  property.value = value;
+
+                  result.computedStyle.add(property);
+                }
+              }
+            });
+      }
+    });
+
+    return result;
+  }
+
+  @ChromeDevtoolsMethod
+  public JsonRpcResult getMatchedStylesForNode(JsonRpcPeer peer, JSONObject params) {
+    final GetMatchedStylesForNodeRequest request = mObjectMapper.convertValue(
+        params,
+        GetMatchedStylesForNodeRequest.class);
+
+    final GetMatchedStylesForNodeResult result = new GetMatchedStylesForNodeResult();
+
+    result.matchedCSSRules = new ArrayList<>();
+
+    final RuleMatch match = new RuleMatch();
+    match.matchingSelectors = new ArrayList<>();
+
+    Selector selector = new Selector();
+    selector.value = "<this_element>";
+
+    match.matchingSelectors.add(0);
+
+    CSSRule rule = new CSSRule();
+
+    rule.origin = "regular";
+    rule.selectorList = new SelectorList();
+
+    rule.selectorList.selectors = new ArrayList<>();
+    rule.selectorList.selectors.add(selector);
+
+    rule.style = new CSSStyle();
+    rule.style.cssProperties = new ArrayList<>();
+
+    match.rule = rule;
+
+    rule.style.shorthandEntries = new ArrayList<>();
+
+    mDocument.postAndWait(new Runnable() {
+      @Override
+      public void run() {
+        mDocument.getElementStyle(
+            mDocument.getElementForNodeId(request.nodeId),
```

Comment:
if the node ID doesn't exist, this will most likely crash hard on an NPE, right? this is probably another good reason we might want to just not have `Document.getElementStyle`